creator-node/scripts/updateContentBlacklist.js

```diff
@@ -1,128 +1,310 @@
   }
 
   return resp.data
 }
 
+/**
+ * 1. Get all blacklisted content
+ * 2. Iterate through passed in CLI args against fetched content
+ * @param {string} type
+ * @param {(number[]|string[])} values cids or ids
+ * @returns {string[]} all segments associated with input
+ */
+async function verifyWithBlacklist ({ type, values, action }) {
+  let allSegments = await getSegments(type, values)
+
+  // Hit creator node /ipfs/:CID route to see if segment is blacklisted
+  let checkFn, filterFn
+  switch (action) {
+    case 'ADD':
+      checkFn = checkIsBlacklisted
+      filterFn = status => !status
+      break
+    case 'REMOVE':
+      checkFn = checkIsNotBlacklisted
+      filterFn = status => status
+      break
+  }
+
+  // Batch requests
+  let creatorNodeResponses = []
+  const checkSegmentBlacklistStatusRequests = allSegments.map(segment => checkFn(segment))
+  for (let i = 0; i < allSegments.length; i += REQUEST_CONCURRENCY_LIMIT) {
+    const creatorNodeResponsesSlice = await Promise.all(checkSegmentBlacklistStatusRequests.slice(i, i + REQUEST_CONCURRENCY_LIMIT))
+    creatorNodeResponses = creatorNodeResponses.concat(creatorNodeResponsesSlice)
+  }
+
+  // Segments that were not accounted for during blacklisting/unblacklisting
+  const unaccountedSegments = creatorNodeResponses
+    .filter(resp => filterFn(resp.blacklisted))
+    .map(resp => resp.segment)
+
+  if (unaccountedSegments.length > 0) {
+    let errorMsg = `Some segments from [${type}] and values [${values}] were not blacklisted/unblacklisted.`
+    errorMsg += `\nNumber of Segments: ${unaccountedSegments.length}`
+    errorMsg += `\nSegments: [${unaccountedSegments.toString()}]`
+    throw new Error(errorMsg)
+  }
+
+  return allSegments
+}
+
+async function getSegments (type, values) {
+  if (type === 'CID') return values
+
+  let allSegments = []
+  let discProvRequests
+  try {
+    // Fetch the segments via disc prov
+    switch (type) {
+      case 'USER': {
+        discProvRequests = values.map(value => axios({
+          url: `${DISCOVERY_PROVIDER_ENDPOINT}/tracks`,
+          method: 'get',
+          params: { user_id: value },
+          responseType: 'json'
+        }))
+        break
+      }
+      case 'TRACK': {
+        discProvRequests = values.map(value => axios({
+          url: `${DISCOVERY_PROVIDER_ENDPOINT}/tracks`,
+          method: 'get',
+          params: { id: value },
+          responseType: 'json'
+        }))
+        break
+      }
+    }
+
+    // Batch requests
+    let discProvResps = []
+    for (let i = 0; i < discProvRequests.length; i += REQUEST_CONCURRENCY_LIMIT) {
+      const discProvResponsesSlice = await Promise.all(discProvRequests.slice(i, i + REQUEST_CONCURRENCY_LIMIT))
+      discProvResps = discProvResps.concat(discProvResponsesSlice)
+    }
+
+    for (const resp of discProvResps) {
+      for (const track of resp.data.data) {
+        for (const segment of track.track_segments) {
```

Comment:
we should always try to avoid double or triple nested loops. for this lets just return segment[0] from each track to do the validation on?

Reply:
I fear that if we only do 1 segment from each track, albeit way faster, will reduce our ability to verify if all the n segments are blacklisted

Reply:
discussed offline - will check every segment, just not using a triple nested for loop 😬